src/core/execute-serialize.c

```diff
@@ -2554,13 +2554,20 @@ static int exec_context_serialize(const ExecContext *c, FILE *f) {
                         return r;
         }
 
-        if (!set_isempty(c->import_credentials)) {
-                char *ic;
-                SET_FOREACH(ic, c->import_credentials) {
-                        r = serialize_item(f, "exec-context-import-credentials", ic);
-                        if (r < 0)
-                                return r;
-                }
+        ExecImportCredential *ic;
+        ORDERED_SET_FOREACH(ic, c->import_credentials) {
+                _cleanup_free_ char *s = NULL;
+
+                s = strdup(ic->glob);
+                if (!s)
+                        return log_oom_debug();
+
+                if (ic->rename && !strextend(&s, " ", ic->rename))
+                        return log_oom_debug();
+
+                r = serialize_item(f, "exec-context-import-credentials", s);
```

Comment:
Just use _format?

Reply:
with `strempty()` for `ic->rename` then?

Reply:
Yes